### PR TITLE
[improvement](multi-catalog) support hive 1.x 

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

The inferface of hive metastore changes from version to version.
Currently, Doris use hive 2.3.7 as hms client version.
When using to connect hive 1.x, some interface such as `get_table_req` does not exist
in hive 1.x. So we can't get metadata from hive 1.x.

In this PR, I copied the HiveMetastoreClient from hive 2.3.7 release, and modify some of interface's
implementation, so that it will use old interface to connect to hive 1.x.

And when creating hms catalog, you can specify the hive version, eg:

```
CREATE CATALOG `hive` PROPERTIES (
  "hive.metastore.uris" = "thrift://127.0.0.1:9083",
  "type" = "hms",
  "hive.version" = "1.1"
);
```

If `hive.version` does not specified, Doris will use hive 2.3.x compatible interface to visit hms.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.86 seconds
 load time: 491 seconds
 storage size: 17122609755 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230112194519_clickbench_pr_79199.html

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java [hunk 1]
```diff
@@ -0,0 +1,2758 @@
...
+     * Capabilities of the current client. If this client talks to a MetaStore server in a manner
+     * implying the usage of some expanded features that require client-side support that this client
+     * doesn't have (e.g. a getting a table of a new type), it will get back failures when the
+     * capability checking is enabled (the default).
+     */
+    public static final ClientCapabilities VERSION = null; // No capabilities.
+    public static final ClientCapabilities TEST_VERSION = new ClientCapabilities(
+            Lists.newArrayList(ClientCapability.TEST_CAPABILITY)); // Test capability for tests.
+
+    ThriftHiveMetastore.Iface client = null;
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java:225]: Could this client support for all different hive versions?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java [hunk 2]
```diff
@@ -0,0 +1,2758 @@
...
+ * https://github.com/apache/hive/blob/rel/release-2.3.7/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
+ * Doris Modification.
+ * To support different type of hive, copy this file from hive repo and modify some method based on hive version
+ * 1. getSchema()
+ * 2. getTable()
+ * 3. tableExists()
+ */
+@Public
+@Unstable
+public class HiveMetaStoreClient implements IMetaStoreClient {
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java:213]: Do we need to modify this file frequently to be consist with community code?

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java:213]: It is copied from hive 2.3.7 release, which is stable

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java (hunk 1 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreClient.java:225]: Yes, this is just a thrift service, and I compare the diff of thrift file among different hive version. Most of service are same.

**Review by Jibing-Li — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**